tests/integration/test_yarn.py

```diff
@@ -14,48 +14,88 @@
         pytest.param(
             utils.TestParameters(
                 repo="https://github.com/cachito-testing/cachi2-yarn-berry.git",
-                ref="4b77c878694e656e7d0ceac6771131273f98f5df",
+                ref="08f09b9340c85fbb84c8fb46fc19a235056a178b",
                 packages=({"path": ".", "type": "yarn"},),
+                check_output=False,
+                check_deps_checksums=False,
                 check_vendor_checksums=False,
-                expected_exit_code=0,
-                expected_output="All dependencies fetched successfully",
+                flags=["--dev-package-managers"],
+                expected_exit_code=2,
+                expected_output="PackageRejected: Yarn zero install detected, PnP zero installs are unsupported by cachi2",
             ),
-            marks=pytest.mark.xfail,  # temporary
             id="yarn_zero_installs",
         ),
+    ],
+)
+def test_yarn_packages(
+    test_params: utils.TestParameters,
+    cachi2_image: utils.ContainerImage,
+    tmp_path: Path,
+    test_data_dir: Path,
+    request: pytest.FixtureRequest,
+) -> None:
+    """
+    Test fetched dependencies for yarn berry.
+
+    :param test_params: Test case arguments
+    :param tmp_path: Temp directory for pytest
+    """
+    test_case = request.node.callspec.id
+
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, f"{test_case}-source", tmp_path
+    )
+
+    utils.fetch_deps_and_check_output(
+        tmp_path, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+
+@pytest.mark.parametrize(
+    "test_params, check_cmd, expected_cmd_output",
+    [
         pytest.param(
             utils.TestParameters(
                 repo="https://github.com/cachito-testing/cachi2-yarn-berry.git",
-                ref="452d7f530513206de63af402536e9ff736c2aa79",
+                ref="70515793108df42547d3320c7ea4cd6b6e505c46",
                 packages=({"path": ".", "type": "yarn"},),
                 check_vendor_checksums=False,
+                flags=["--dev-package-managers"],
                 expected_exit_code=0,
                 expected_output="All dependencies fetched successfully",
             ),
-            marks=pytest.mark.xfail,  # temporary
-            id="yarn_no_zero_installs",
+            [],
+            "",
```

Comment:
We should make sure that the `yarn berryscary` command is executed and that the proper output is generated.

Reply:
From the tests I'm running here, adding `['yarn', 'berryscarry']` as the check_cmd makes the test fail with:
```           
AssertionError: ['yarn', 'berryscary'] command failed, Output: yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Couldn't find a package.json file in "/tmp/yarn_e2e_test-source"
```
I didn't dig deep into it, but maybe the image is falling back to Yarn 1?

Reply:
The default yarn version in the node container images is usually still yarn v1. We'll need to get the correct yarn version installed in the container via another method that doesn't require network access...

corepack... :harold:

Reply:
I thought even yarn 1 would replace itself with https://github.com/cachito-testing/cachi2-yarn-berry/blob/no-zero-installs/.yarn/releases/yarn-3.6.1.cjs, so it would work :thinking:

Reply:
It looks like it's caused by a mismatch in the mounted directory paths at different stages of the test. When the test image is built, the source directory within the container is /tmp/{test_case_name}-source. Later, when you execute a command on the built image, the source directory path has changed to whatever the tmp_path fixture is joined with {test_case_name}-source.

To see what I'm talking about, if you copy the source directory in the Containerfile after running the yarn install commands to somewhere like /src and change the WORKDIR to /src, it should magically start working

Reply:
Well, it works /o\

Reply:
Maybe we shouldn't to mount the tmp_dir into the image when we're running it?

https://github.com/containerbuildsystem/cachi2/blob/06f77cb4b656ae0c22142e392e1d65bb41ccf33c/tests/integration/utils.py#L67

I don't see why the image should rely on the source code mounted from an external dir just to run.